Project3

Flash-Kaa opened the pull request.

Flash-Kaa committed:
init

Flash-Kaa committed:
complited task with http-request

Flash-Kaa committed:
complited

strelchm commented on src/main/java/edu/project3/terminal/Terminal.java:
```diff
@@ -0,0 +1,120 @@
+        scanner = new Scanner(inputStream);
+    }
+
+    public static TerminalRequest expectRequest() {
```
nit: Очень большой метод, наверняка можно разбить на более маленькие

strelchm commented on src/main/java/edu/project3/statistics/PopularHourInfo.java:
```diff
@@ -0,0 +1,29 @@
+        }
+
+        table.add(new Pair("Час", "Количество запросов"));
+        for (Integer i : map.keySet()) {
```
nit: итерироваться можно по entrySet()

strelchm commented on src/test/java/edu/project3/ReadersTest.java:
```diff
@@ -0,0 +1,139 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ReadersTest {
```
При запуске теста ошибки при парсинге дат некоторых строк
![image](https://github.com/Flash-Kaa/Java-Course-Tinkoff/assets/23243577/7ec7b5d0-9f52-43a8-88aa-8a55c78286f2)

strelchm replied:
из-за этого тесты проходят, но отчеты пустые
![image](https://github.com/Flash-Kaa/Java-Course-Tinkoff/assets/23243577/db8ba485-91b0-4832-94ae-2a8428a3deb4)

strelchm replied:
видимо так как у меня месяц ожидается в кириллице, напр. 
`System.out.println(DATE_TIME_FORMATTER.format(LocalDateTime.now()));`
выводит
`07/янв./2024:17:01:44`
Подумать, как такое ограничение можно обойти

strelchm commented on src/main/java/edu/project3/LoggerAnalyzer.java:
```diff
@@ -0,0 +1,60 @@
+                return sb.toString();
+            }
+        );
+        thread = new Thread(task);
```
В API фьюч специальные методы как для асинхронного, так и блокирующего запуска

strelchm replied:
Можно посмотреть в сторону `CompletableFuture.runAsync` / `CompletableFuture.supplyAsync`

strelchm commented on src/main/java/edu/project3/terminal/Terminal.java:
```diff
@@ -0,0 +1,120 @@
+            try {
+                uri = new URI(commands[curInd++]);
+            } catch (URISyntaxException e) {
+                LOGGER.error(String.format("Incorrect URI: %s", commands[curInd - 1]));
```
nit: `LOGGER.error("Command '{}' not found, did you mean '{}'", actual, expected);`

strelchm reviewed (approved):
logs.txt лучше вынести в тестовые ресурсы

Flash-Kaa committed:
refactoring
fix: parse dateTime
change: logs.txt in resources
change: FutureTask without "new Thread()"
